### ISSUE TITLE
certificate_map support

### ISSUE THREAD

**Issue by tbjorkna** [terraform-google-modules/terraform-google-lb-http#267]
### TL;DR

If supported already, I wonder how to use `certificate_map` with this module (serverless_negs)? Specifically [this](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_target_https_proxy#certificate_map). Essentially the last step in [this guide](https://cloud.google.com/certificate-manager/docs/deploy-google-managed-dns-auth#attach_the_certificate_map_to_the_target_proxy) is what I'm attempting to get done with the module.

Or is this supposed to be done through the resource `google_certificate_manager_certificate_map` using the `gclb_targets` block and `target_https_proxy` according to this [link](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/certificate_manager_certificate_map#nested_gclb_targets).

### Terraform Resources

```markdown
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_target_https_proxy#certificate_map
```

**Comment by hollow** [comment]
I have added support for `certificate_map` in my fork: https://github.com/hollow/terraform-google-lb-http/commit/4e9f4971ddb9793fe4bedad3ac4380baa451a91b

This fork also contains #269 and support for `optional()` variables in backend configs to get rid of the need to specify most variables as `null` in the module call. After #269 has been merged I can prepare a proper PR for `certificate_map` support.

**Comment by charims** [comment]
@hollow looks like #269 got merged, I am excited to get certifficate_map capability added into the module.

**Comment by hollow** [comment]
@charims I'm quite busy these days, but I'll see what I can do. If anyone wants to backport certificate map support from my fork, please do! 🙏
